src/main/java/hudson/remoting/Callable.java

```diff
@@ -27,6 +27,7 @@
 
 import java.io.Serializable;
 
+//TODO: Make it SerializableOnlyOverRemoting?
```

Comment:
I was going to say that this marker should be applied only to concrete implementations which actually define a `writeReplace` etc. that assumes Remoting, since there could be a `NotReallyRoleSensitiveCallable` which is serialized for some other purpose. But then I saw various callable classes which need the current channel, for which having access to the `default` method would be useful. Anyway IIUC there is no actually harm done in implementing this interface but then serializing the object in some other way, provided you do not actually call the `default` method.

BTW if you mark it here, do so in core also on `FileCallable`.